### PR TITLE
[CN-846] Reflect bucket connection errors to the Operator CRs

### PR DESCRIPTION
**Pull request by semihbkgr**
## User Impact

Reflect bucket connection related errors from agent to the Operator CRs

### CODE REVIEW

**Comment by semihbkgr** [comment]
The new added [e2e test](https://github.com/hazelcast/hazelcast-platform-operator/pull/828/files#diff-7f7171817eeded43d92d903017408a21e4d3597f7d47205e9247f9a2938ce8ab) have passed with the new version of agent. 👍
http://reportboard.s3-website-us-east-1.amazonaws.com/pr/3303/#behaviors/b1a8273437954620fa374b796ffaacdd/396b20323332fa31/

**Review by hasancelik — APPROVED** [review]

File: test/e2e/hazelcast_backup_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		if !ee {
+			Skip("This test will only run in EE configuration")
+		}
+		setLabelAndCRName("hb-1")
+		ctx := context.Background()
+		clusterSize := int32(1)
+		var pvcSizeInMb = 1
+		var bucketURI = "gs://operator-e2e-external-backup"
+		var secretName = "br-incorrect-secret-gcp"
+		var credential = `{
```
Comment (SeriyBg) [inline test/e2e/hazelcast_backup_test.go:52]: I would suggest not using backquotes raw strings. IMO, this indenting make it harder to read

**Review by SeriyBg — APPROVED** [review]
One small neat

File: test/e2e/hazelcast_backup_test.go (hunk 1 above)
Comment (semihbkgr) [reply test/e2e/hazelcast_backup_test.go:52]: The only purpose of using backquotes is eliminating the backslashes in the string.

Comment (SeriyBg) [reply test/e2e/hazelcast_backup_test.go:52]: Is it needed at all? Since it's JSON, it should be valid anyway, isn't it?

Comment (SeriyBg) [reply test/e2e/hazelcast_backup_test.go:52]: Can we, then, indent it to make it more readable?

**Pull request merged**